src/overlays/actors/ovl_En_Zf/z_en_zf.c

```diff
@@ -82,115 +151,2291 @@ static ColliderQuadInit D_80B4A204 = {
+                this->hopAnimIndex = 0;
+            }
+
+            if ((this->unk_408 != 0.0f) || (this->unk_40C != 0.0f)) {
+                this->hopAnimIndex = 1;
+            }
+
+            lastFrame = Animation_GetLastFrame(sHoppingAnims[this->hopAnimIndex]);
+
+            switch (this->hopAnimIndex) {
+                case 0:
+                    this->actor.velocity.y = 0.0f;
+                    this->actor.world.pos.y = this->actor.floorHeight;
+                    break;
+
+                case 1:
+                    this->actor.velocity.y = this->unk_40C + 10.0f;
+                    this->actor.speedXZ = this->unk_408;
+                    this->unk_408 = 0.0f;
+                    this->unk_40C = 0.0f;
+                    break;
+
+                case 2:
+                    this->actor.world.pos.y = this->actor.floorHeight;
+                    lastFrame = 3.0f;
+                    break;
+
+                default:
+                    break;
+            }
+
+            Animation_Change(&this->skelAnime, sHoppingAnims[this->hopAnimIndex], 1.5f, 0.0f, lastFrame, 2, 0.0f);
+        }
+
+        if ((globalCtx->gameplayFrames & 0x5F) == 0) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_RIZA_CRY);
+        }
+    }
+}
+
+void EnZf_SetupHopAway(EnZf* this, GlobalContext* globalCtx) {
+    this->hopAnimIndex = 0;
+    Animation_PlayOnce(&this->skelAnime, sHoppingAnims[0]);
+    this->action = ENZF_ACTION_HOP_AWAY;
+    this->curPlatform = EnZf_FindPlatform(&this->actor.world.pos, this->curPlatform);
+    this->nextPlatform =
+        EnZf_FindNextPlatformAwayFromPlayer(&this->actor.world.pos, this->curPlatform, this->homePlatform, globalCtx);
+    EnZf_SetupAction(this, EnZf_HopAway);
+}
+
+void EnZf_HopAway(EnZf* this, GlobalContext* globalCtx) {
+    f32 sp74;
+    f32 sp70 = 1.0f;
+    f32 phi_f20 = 550.0f;
+    s32 pad;
+    f32 phi_f20_2;
+    f32 phi_f0;
+    s32 pad2;
+    s16 sp5A;
+    s32 sp54;
+    s32 temp_v1_2;
+    s32 phi_v1;
+
+    sp74 = Actor_WorldDistXZToPoint(&this->actor, &sPlatformPositions[this->nextPlatform]);
+    sp54 = this->hopAnimIndex;
+
+    if ((globalCtx->gameplayFrames & 0x5F) == 0) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_RIZA_CRY);
+    }
+
+    // Upstairs
+    if (this->actor.world.pos.y >= 420.0f) {
+        phi_f20 = 280.0f;
+    }
+
+    sp5A = Actor_WorldYawTowardPoint(&this->actor, &sPlatformPositions[this->nextPlatform]);
+
+    switch (this->hopAnimIndex) {
+        case 0:
+            this->actor.world.rot.y = sp5A;
+            this->actor.shape.rot.y = sp5A + 0x8000;
+            D_80B4AB30 = 0;
+            this->homePlatform = this->curPlatform;
+            temp_v1_2 = !EnZf_PrimaryFloorCheck(this, globalCtx, 107.0f);
+            temp_v1_2 |= !EnZf_PrimaryFloorCheck(this, globalCtx, 220.0f) << 1;
+            this->hopAnimIndex++;
+
+            switch (temp_v1_2) {
+                case 1:
+                case 1 | 2:
```

Comment:
```suggestion
                case 0b01:
                case 0b11:
```
may make sense to use here

Reply:
Hm, hadn't thought of that. I don't know if this sort of thing happens elsewhere in the codebase.

Reply:
Not that I know of. there also isn't a single use of the `0b` binary literal anywhere

Reply:
IDO may not even support this: https://stackoverflow.com/questions/18244726/why-doesnt-c-have-binary-literals

Reply:
And it doesn't:
```
cfe: Error: src/overlays/actors/ovl_En_Zf/z_en_zf.c, line 2405: Syntax Error
                     case 0b11  :
 --------------------------^
```